### PR TITLE
test: add consumer e2e test

### PR DESCRIPTION
**Pull request by idbeta**
add consumer e2e test example
add public method "PartialBody"

### CODE REVIEW

**Comment by idbeta** [comment]
ping @nic-chen @membphis

File: api/test/e2e/consumer_test.go
Comment (moonming) [inline api/test/e2e/consumer_test.go:73]: Need add test cases for non-auth header

File: api/test/e2e/consumer_test.go
Comment (moonming) [inline api/test/e2e/consumer_test.go:100]: Not test data plane

File: api/test/e2e/consumer_test.go
Comment (moonming) [inline api/test/e2e/consumer_test.go:171]: Ditto

File: api/test/e2e/consumer_test.go
Comment (moonming) [inline api/test/e2e/consumer_test.go:186]: Why create again?

File: api/test/e2e/consumer_test.go
Comment (moonming) [inline api/test/e2e/consumer_test.go:201]: Ditto

**Review by moonming — CHANGES_REQUESTED** [review]

File: api/test/e2e/consumer_test.go
Comment (idbeta) [reply api/test/e2e/consumer_test.go]: Does not need to be created again here.

File: api/test/e2e/consumer_test.go
Comment (idbeta) [reply api/test/e2e/consumer_test.go:81]: I added non-auth header test in the new submit
```
{
  caseDesc:     "verify route without auth",
  Object:       APISIXExpect(t),
  Method:       http.MethodGet,
  Path:         "/hello",
  ExpectStatus: http.StatusUnauthorized,
  Sleep:        sleepTime, 
}
```

File: api/test/e2e/consumer_test.go
Comment (idbeta) [reply api/test/e2e/consumer_test.go:117]: I will add test data plane here. you can see the new submit.

File: api/test/e2e/consumer_test.go
Comment (idbeta) [reply api/test/e2e/consumer_test.go:188]: My purpose  is to prevent the two use cases from coupling.

File: api/test/e2e/consumer_test.go
Comment (idbeta) [reply api/test/e2e/consumer_test.go]: My purpose  is to prevent the two use cases from coupling.

File: api/test/e2e/consumer_test.go
Comment (membphis) [inline api/test/e2e/consumer_test.go:39]: bad indentation

File: api/test/e2e/consumer_test.go
Comment (membphis) [inline api/test/e2e/consumer_test.go:80]: we need to delete the consumer `jack` at the end of the test case

File: api/test/e2e/consumer_test.go
Comment (membphis) [inline api/test/e2e/consumer_test.go:102]: bad indentation too

File: api/test/e2e/consumer_test.go
Comment (membphis) [inline api/test/e2e/consumer_test.go:214]: remove those blank line

File: api/test/e2e/consumer_test.go
Comment (membphis) [reply api/test/e2e/consumer_test.go:80]: it may cause wrong for other test cases

Comment (idbeta) [reply api/test/e2e/consumer_test.go:80]: I delete the created data uniformly in the last function `TestConsumer_teardown`

File: api/test/e2e/consumer_test.go
Comment (idbeta) [reply api/test/e2e/consumer_test.go]: fixed

File: api/test/e2e/consumer_test.go
Comment (idbeta) [reply api/test/e2e/consumer_test.go]: fixed

File: api/test/e2e/consumer_test.go
Comment (idbeta) [reply api/test/e2e/consumer_test.go]: fixed

**Comment by LiteSun** [comment]
There's a conflict here. Please fix it. @idbeta

**Comment by idbeta** [comment]
> There's a conflict here. Please fix it. @idbeta

fixed.

**Review by nic-chen — APPROVED** [review]

**Comment by juzhiyuan** [comment]
The CD failure is not related to this PR, according to @nic-chen's approval, I will merge this PR soon.

**Comment by idbeta** [comment]
relate https://github.com/apache/apisix-dashboard/issues/607

File: api/test/e2e/consumer_test.go
Comment (moonming) [inline api/test/e2e/consumer_test.go:201]: Why use admin api to do controller plane job?

**Comment by moonming** [comment]
@nic-chen @membphis please take a look. Can we use admin API in e2e test?

**Comment by juzhiyuan** [comment]
This PR would be reverted to https://github.com/apache/apisix-dashboard/pull/829 , one PR should be merged only when there have 2 approval at least.

**Comment by nic-chen** [comment]
> @nic-chen @membphis please take a look. Can we use admin API in e2e test?

no, we can't. my fault, I shouldn't approve it.

**Pull request merged**
